rotkehlchen/tests/unit/test_assets.py

```diff
@@ -541,7 +541,15 @@ def test_cryptocompare_asset_support(cryptocompare):
         'HAT',  # Top Hat but CC has Hawala.Exchange
 
     )
-    for asset_data in GlobalDBHandler.get_all_asset_data(mapping=False):
+    # Get all asset identifiers and check each one using the existing get_asset_data method
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        cursor.execute('SELECT identifier FROM assets')
+        all_identifiers = [row[0] for row in cursor]
+
+    for identifier in all_identifiers:
+        asset_data = GlobalDBHandler.get_asset_data(identifier, form_with_incomplete_data=False)
```

Comment:
i get that this is a test but this is expensive as you can get the data once. it can be a test utility function so you can reuse it below.